dffml/high_level/ml.py

```diff
@@ -51,6 +51,19 @@ async def train(model, *args: Union[BaseSource, Record, Dict[str, Any]]):
     >>>
     >>> asyncio.run(main())
     """
+    if any(isinstance(arg, list) for arg in args):
+        if isinstance(model.config.predict, Features):
+            args = list_records_to_dict(
+                [feature.name for feature in model.config.features]
+                + [feature.name for feature in model.config.predict],
+                *args,
+            )
+        else:
+            args = list_records_to_dict(
+                [feature.name for feature in model.config.features]
+                + [model.config.predict.name],
+                *args,
+            )
     async with contextlib.AsyncExitStack() as astack:
         # Open sources
         sctx = await astack.enter_async_context(records_to_sources(*args))
```

Comment:
Let's make model a keyword argument and if it's present, then we can support converting lists to Record instances. We should raise an exception if we were unable to convert a list to a Record.

```suggestion
        sctx = await astack.enter_async_context(records_to_sources(*args, model=model))
```

Reply:
Assuming this was supposed to be for the function `list_records_to_dict()`